scripts/analysis/get-all-components.ts

```diff
@@ -0,0 +1,91 @@
+import { readdirSync, readFileSync } from 'fs';
+import {
+	packageNames,
+	getComponentPackageNamesWithPaths,
+} from '../package-names';
+import { paths, getKitchenComponentPaths } from '../paths';
+import { parse } from '@typescript-eslint/typescript-estree';
+
+const getPackageComponents = (path: string | string[]): string[] => {
+	if (Array.isArray(path)) {
+		return path.flatMap((p) => getPackageComponents(p));
+	}
+
+	const contents = readdirSync(path);
+
+	if (contents.includes('index.tsx')) {
+		return getExportsFromFile(`${path}/index.tsx`);
+	} else {
+		return contents
+			.filter(
+				(item) =>
+					item.endsWith('.tsx') && !item.endsWith('stories.tsx'),
```

Comment:
if you swap these the `stories.tsx` check can exit sooner

Reply:
Done in https://github.com/guardian/source/pull/972/commits/30305b6a42724d715c545e46c13e62450efff73a